src/analysis/lattice.h

```diff
@@ -0,0 +1,27 @@
+#ifndef wasm_analysis_lattice_h
+#define wasm_analysis_lattice_h
+
+#include <bitset>
+#include "wasm.h"
+
+namespace wasm::analysis {
+
+enum LatticeComparison {
+    NO_RELATION,
+    EQUAL,
+    LESS,
+    GREATER
+};
+
+template<size_t N>
```

Comment:
In general the size of the set may not be known at compile time (for example if it is a set of local variables, we don't know how many there will be until we know what function we are analyzing), so actually I think using `std::vector<bool>` would be better than using a `bitset` after all.

(Alternatively, if this lattice is just to prove out the infrastructure rather than for near-term use in a real analysis, it would be fine to keep `N` a compile time constant)

Reply:
This is true. The `bitset` powerset lattice is only left here just to prove out the infrastructure. I'm leaving changing it to a dynamically sized powerset lattice in a future PR.